README.md

```diff
@@ -1,3 +1,19 @@
-## Iceberg Docs
+# Apache Iceberg Documentation Site
 
-This repository contains the markdown documentation hosted at https://iceberg.apache.org.
+This repository contains the documentation for [Apache Iceberg](https://github.com/apache/iceberg).
+It's built with [Hugo](https://gohugo.io/) and hosted on GitHub pages.
```

Comment:
hosted at https://iceberg.apache.org ?

Reply:
Yeah, I think that it is actually hosted by ASF infra.

Reply:
Fixed!